### PR TITLE
redirect asm_moe_tkw1 call to fused_moe in order to force kernel tuning

### PR DESCRIPTION
**Pull request by antsaukk**
## Motivation

This change introduces merge of kernel selection API `asm_moe_tkw1` from `fused_moe_bf16_asm.py` into `fused_moe.py`. The change is required in order to unlock kernel selection logic for `tkw1` case from predefined tuned_fmoe.csv config, which brings performance improvement for low concurrency cases in top tier customer model. 

## Technical Details

In order to unlock kernel selection functionality from tuned_fmoe.csv config for `asm_moe_tkw1`, the function call to `asm_moe_tkw1` is forwarded into `fused_moe`. The selection logic of `asm_moe_tkw1` has been left unchanged but is extracted into a new function `fused_moe_stage1_tkw1` under `fused_moe.py` module. The functions' interfaces have been modified accordingly to accommodate new arguments.

## Test Plan

The functionality has been tested with `aiter/op_tests` to verify that selection logic works as intended. In addition to that, tests within the model were conducted to confirm end-to-end correctness.

## Test Result

Above mentioned tests confirm that correctness is preserved.

## Submission Checklist

At least those test cases should be verified: 

1. `AITER_LOG_MORE=1 python ./op_tests/test_moe_tkw1.py -m 1 -dim 5120 -hdim 1024 -e 128 -d bf16`, uses 1 stage fmoe; selects 32x64 kernel 
2. `AITER_LOG_MORE=1 python ./op_tests/test_moe_tkw1.py -m 17 -dim 5120 -hdim 1024 -e 128 -d bf16`, uses 1 stage fmoe; selects 32x128 kernel 
3. `AITER_LOG_MORE=1 python ./op_tests/test_moe_tkw1.py -m 300 -dim 5120 -hdim 1024 -e 128 -d bf16`, uses 1 stage fmoe;  selects 32x128 kernel 
4. `AITER_LOG_MORE=1 python ./op_tests/test_moe_2stage.py -d bf16 -dim 5120,1024 -t 1 -q 2 -a silu -e 128 -k 1` uses 2 stage moe, with 32x512 asm kernel and ck_moe_stage2 selected
5.   `AITER_LOG_MORE=1 python ./op_tests/test_moe_2stage.py -d bf16 -dim 5120,1024 -t 17 -q 2 -a silu -e 128 -k 1` uses 2 stage moe, with 32x512 asm kernel and ck_moe_stage2 selected

## Additional notes 

Please note that files
hsa/gfx950/fmoe/gelu/fmoe_bf16_pertokenFp8_g1u1_gelu_tkw1.csv, hsa/gfx950/fmoe/silu/fmoe_bf16_pertokenFp8_g1u1_silu_tkw1.csv,
hsa/gfx950/fmoe/silu/fmoe_bf16_pertokenInt8_g1u1_tkw1_silu_32x64.co,
hsa/gfx950/fmoe/gelu/fmoe_bf16_pertokenInt8_g1u1_tkw1_gelu_32x64.co, 
aiter/configs/tuned_fmoe.csv 

are not meant to be merged in this PR. However, they are required here in order to verify correctness functionality. Upon approvement, they will be removed and merger in separate [PR](https://github.com/ROCm/aiter/pull/1386).

### CODE REVIEW

File: aiter/fused_moe.py [hunk 1]
```diff
@@ -429,6 +454,156 @@ def fused_moe_1stage(
...
+                a8_scale = torch.empty(M, dtype=dtypes.fp32, device=device)
+                fc1_smooth_scale = torch.ones(
+                    model_dim, dtype=dtypes.fp32, device=device
+                )
+                aiter.smoothquant_fwd(a8, hidden_states, fc1_smooth_scale, a8_scale)
+            else:
+                logger.warning("FMOE fall into pure torch quant...")
+                a8, a8_scale = aiter.pertoken_quant(hidden_states, quant_dtype=w1.dtype)
+        if w2.shape[2] * 2 * lastdim_mul == w1.shape[1]:
+            fmoe_func = aiter.fmoe_g1u1_tkw1
```
Comment (valarLip) [inline aiter/fused_moe.py]: this the only one for tkw1, we don't real need the whole "def fused_moe_stage1_tkw1("

Comment (antsaukk) [reply aiter/fused_moe.py]: Could you please check with the latest changes?

File: aiter/fused_moe.py [hunk 2]
```diff
@@ -618,6 +647,19 @@ def FinalFunc():
                 run_1stage = token > 32
             elif q_type != QuantType.per_1x32:
                 run_1stage = token < 256
+        elif (
```
Comment (valarLip) [inline aiter/fused_moe.py]: why add elif instead of remove https://github.com/ROCm/aiter/pull/1366/files#diff-b34aed242c551a47f93071f37cdb3552291c402b6a66cfa3a2a0fbe1a4ce0f10L604

File: aiter/fused_moe.py
Comment (valarLip) [inline aiter/fused_moe.py]: let's add one more column for doweight_stage1

**Review by valarLip — APPROVED** [review]
thanks your contribution, LGTM

File: aiter/fused_moe.py [hunk 3]
```diff
@@ -640,6 +668,7 @@ def FinalFunc():
         f"[fused_moe] using {'1stage' if run_1stage else '2stage'} {'default' if cfg is None else tag} for {keys} "
     )
     if run_1stage:
+        logger.info(f"[get_2stage_cfgs] run_1stage")
```
Comment (valarLip) [inline aiter/fused_moe.py]: i would like remove this log

Comment (antsaukk) [reply aiter/fused_moe.py]: Done
